### ISSUE TITLE
Add tracing

### ISSUE THREAD

**Issue by ogxd** [aerospike/aerospike-client-csharp#79]
# Context

Tracing is a very powerful way concept for modern backends and found its way as a basic of observability as logs and metrics are today.
In dotnet, distributed tracing was pushed forward in .NET 6 with the introduction of the System.Diagnostics.Activity APIs in the base class library. 
See [this article](https://jimmybogard.com/activity-enrichment-in-asp-net-core-6-0/) and [the official documentation](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity?view=net-7.0).    
Instrumentation was progressively introduced for the most common protocols over the last two years, starting with the HttpClient and following with Grpc.Net.Client (see [some example from the source code](https://github.com/grpc/grpc-dotnet/blob/456098bcfeb9c796c19ea023b3ab3703181f1089/src/Shared/TelemetryHeaderHandler.cs#L94)).    
Given the critical place aerospike spike has in many backends, it would me more than welcome for this client to implement the Activity APIs.

# How
The `Activity` APIs is quite barebone and gives a lot of freedom as of how to to use it. OpenTelemetry (part of the CNCF) proposes some naming conventions, which are already widely adopted by the industry (Grpc.Net.Client uses them).

- The base idea is to create a new `Activity` whenever there is an I/O (so in the case of retries, it means several Activities will be created).
- Tags must be added to add some information such as: which node was called? What kind of operation is it? What is the key? Is it the first attempt? ...
- `Activity.Current` is uses `AsyncLocal`. It's an important detail to understand how to API works under the hood
- In case of error, the ongoing `Activity` state must be marked as error
- When there is no listener, `ActivitySource.StartActivity(...)` will return null. That means that this won't affect performances in any way if there is no trace/listener ongoing. That means that it will make no difference for people that don't use tracing.

**Comment by ogxd** [comment]
See PR #79

**Comment by shannonklaus** [comment]
We like this idea. We will put it in our next release

**Comment by ogxd** [comment]
Awesome!    
Here is an example of what it looks like with the PR (we have been testing this fork on our end).   
<img width="1539" alt="image" src="https://user-images.githubusercontent.com/12692438/229123465-5ccec2f3-acd2-46b7-90f6-31bc157309d2.png">
This UI is from [Grafana Tempo](https://github.com/grafana/tempo) but it would be similar in all other solutions for visualizing traces that are support the [OpenTelemetry semantic conventions](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/) (Datadog APM, Jaeger, LightStep, Zipkin, ...)

**Comment by guillaumed-unity** [comment]
That's pretty cool, any plan to include this >? @shannonklaus ?

**Comment by shannonklaus** [comment]
Yes, I am still planning on including this at some point, it has been out prioritized by other features, but it is still on my radar
